specification/README.md

```diff
@@ -366,28 +367,60 @@ Since function definitions are reusable, their data input parameters are defined
             "type": "string",
             "description": "Context attribute name of the CloudEvent which value is to be used for event correlation"
         },
+        "kind": {
+            "type" : "string",
+            "enum": ["consumed", "produced"],
+            "description": "Defines the events as either being consumed or produced by the workflow. Default is consumed",
+            "default": "consumed"
+        }, 
         "metadata": {
           "$ref": "#/definitions/metadata",
           "description": "Metadata information"
         }
     },
-    "required": ["name", "source", "type"]
+    "if": {
+        "properties": {
+          "kind": {
+            "const": "consumed"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "name",
+          "source",
+          "type"
+        ]
+      },
+      "else": {
+        "required": [
+          "name",
+          "type"
+        ]
+     }
 }
 ```
 
 </details>
 
 Defines events that can be consumed or produced during workflow execution.
-Consumed events can trigger actions to be executed. Events can also be produced during workflow
-execution to be consumed by clients.
 
-As serverless workflow definitions are vendor neutral, so should be the events definitions that they consume and produce.
-As such event format within serverless workflows uses the [CloudEvents](https://github.com/cloudevents/spec) specification to describe events.
+Consumed events can trigger creations of workflow instances, or continue execution of existing workflow instances.
+Workflows can produce events during their execution. These events can be then consumed by clients (or other workflows).
 
```

Comment:
Please clarify if an eventdef can serve for both consumed and produced event types or if that case requires a second eventdef.

Reply:
as defined currently it cannot. and as you mentioned i think the use cases where this happens seem very small in which case i think it would be ok. I will update the spec doc to mention this clearly.